### PR TITLE
feature/330 - Allow hyperlinking to a specific weapon or character or (future nft)

### PR DESCRIPTION
**Pull request by FilipBartos**
### All Submissions

* [x] Can you post a screenshot of your changes (if applicable)?

NFT Listing

image

NFT share action

image

NFT copied link result

image

### New Feature Submissions

* [x] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?

https://github.com/CryptoBlades/cryptoblades/issues/330

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

(insert description here)

### CODE REVIEW

**Comment by seiyria** [comment]
Can we get the share button to the left of the clear filters button? Having it on every weapon is a bit much. With it being there, it will allow sharing a set of market results better. We can add a share id option on the search nfts page, since that one is look up by id.

File: frontend/src/router.ts [hunk 1]
```diff
@@ -32,7 +32,8 @@ function createRouter() {
   let marketRoutes: RouteConfig[] = [];
   if(featureFlagMarket) {
     marketRoutes = [
-      { path: '/market', name: 'market', component: Market }
+      { path: '/market/:category/:type/:id', name: 'marketSearch', component: Market },
```
Comment (seiyria) [inline frontend/src/router.ts:35]: this should really all be query params on /market. we have a lot of search criteria to go through.

**Comment by FilipBartos** [comment]
@seiyria sure, what about separate view - something like "seller's profile" that would list characters/weapons from specified seller address? Something like at the screens below:

image
image

**Comment by seiyria** [comment]
Sure, we could do something like that. What you'll want to do is accept query parameters for everything that goes into the search query - seller address, element, minlevel, maxlevel, etc, and use that. then people can share their current settings much more easily than having more market routes.

**Review by seiyria — CHANGES_REQUESTED** [review]
conflicts need to be fixed, as well as the other change I requested

File: frontend/src/views/Seller.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  },
+
+  methods: {
+    ...(mapActions(['fetchMarketNftIdsBySeller' ])),
+
+    async fetchData () {
+      const route = this.$route;
+      const sellerAddress = route.params.sellerAddress;
+      this.items = await this.searchListingsBySellerThroughChain(sellerAddress);
+      console.warn(this.items);
```
Comment (seiyria) [inline frontend/src/views/Seller.vue:68]: please remove console statements

**Comment by ultradeq** [comment]
Hi @seiyria , what about placing the "share market link" (or something like "share my market") to the character display area (see attached image)? It would then copy the url .../market/seller/:accountAddress (where the user's items for sale would be listed) to the clipboard.
cb-share-market-link

**Comment by seiyria** [comment]
I think that location would be fine, but again, everything should be a query parameter. There should be no new routes for this.

**Comment by kcper** [comment]
As there were no updates here, I resolved the issue using "..." menu with "Copy link" option and NFT Display component here: https://github.com/CryptoBlades/cryptoblades/pull/851

**Pull request closed without merging**